http2/src/client_conn.rs

```diff
@@ -103,23 +103,29 @@ impl LoopInner for ClientInner {
                                .map_err(HttpError::CompressionError).unwrap();
         let headers: Vec<Header> = headers.into_iter().map(|h| Header::new(h.0, h.1)).collect();
 
-        let mut stream: &mut HttpClientStream = match self.common.get_stream_mut(frame.get_stream_id()) {
+        let result = match self.common.get_stream_mut(frame.get_stream_id()) {
             None => {
                 // TODO(mlalic): This means that the server's header is not associated to any
                 //               request made by the client nor any server-initiated stream (pushed)
                 return;
             }
-            Some(stream) => stream,
-        };
-        // TODO: hack
-        if headers.len() != 0 {
-
-            if let Some(ref mut response_handler) = stream.response_handler {
-                response_handler.send(ResultOrEof::Item(HttpStreamPart {
-                    content: HttpStreamPartContent::Headers(headers),
-                    last: frame.is_end_of_stream(),
-                })).unwrap();
+            Some(ref mut stream) if headers.len() != 0 => {
```

Comment:
`ref` for cannot bind by-move into a pattern guard.
`mut` for cannot borrow data mutably in a `&` reference.